pkg/util/fedinformer/transform.go

```diff
@@ -7,11 +7,16 @@ import (
 	corev1 "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/api/meta"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/tools/cache"
 )
 
 // StripUnusedFields is the transform function for shared informers,
 // it removes unused fields from objects before they are stored in the cache to save memory.
 func StripUnusedFields(obj interface{}) (interface{}, error) {
+	if tombstone, ok := obj.(cache.DeletedFinalStateUnknown); ok {
+		obj = tombstone.Obj
+	}
```

Comment:
Could you please add a comment on this to explain the reason why we need to check the state? I'm afraid not many people know the details.

Reply:
I will add a comment as follow, is that ok? @RainbowMango 
// This could happen if watch deletion event was missed while disconnected from apiserver

Reply:
We have lessons learned on this, such as #3478 fixing a similar issue.
I'll go over the code base and check if we missed it somewhere and add the comments accordingly. (It's not fair asking you to do this in this PR.).